app/models/user.rb

```diff
@@ -26,11 +27,32 @@ def new_token
     end
   end
 
-  def authenticated? remember_token
-    BCrypt::Password.new(remember_digest).is_password?(remember_token)
+  def authenticated? attribute, token
+    digest = send "#{attribute}_digest"
+    return false if digest.nil?
+    BCrypt::Password.new(digest).is_password? token
   end
 
   def forget
     update_attribute :remember_digest, nil
   end
+
+  def activate
+    update_attributes activated: true, activated_at: Time.zone.now
+  end
+
+  def send_activation_email
+    UserMailer.account_activation(self).deliver_now
```

Comment:
I have some question here. 
1. Why you send email to activate email ? 
2. When a user click on the email and come back to the application, how the application verify that user and allow him to login.

Reply:
I send activate email to verify realistic user.
When user click on the link include token in the activate email, app find user by user' email and authenticate the token by comparing with the activation digest.If the user is authenticated, change the activated status from false to true.

Reply:
@maiq721 :1st_place_medal: :100: :+1: